lib/src/issue.rs

```diff
@@ -0,0 +1,128 @@
+// git-dit - the distributed issue tracker for git
+// Copyright (C) 2016, 2017 Matthias Beyer <mail@beyermatthias.de>
+// Copyright (C) 2016, 2017 Julian Ganz <neither@nut.email>
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+
+//! Issues
+//!
+//! This module provides the `Issue` type and related functionality.
+//!
+
+use git2::{self, Oid, Reference, References};
+use std::fmt;
+use std::result::Result as RResult;
+
+use error::*;
+use error::ErrorKind as EK;
+
+
+/// Issue handle
+///
+/// Instances of this type represent single issues. Issues reside in
+/// repositories and are uniquely identified by an id.
+///
+pub struct Issue<'r> {
+    repo: &'r git2::Repository,
+    id: Oid,
+}
+
+impl<'r> Issue<'r> {
+    /// Create a new handle for an issue with a given id
+    ///
+    pub fn new(repo: &'r git2::Repository, id: Oid) -> Self {
+        Issue { repo: repo, id: id }
+    }
+
+    /// Get the issue's id
+    ///
+    pub fn id(&self) -> Oid {
+        self.id
+    }
+
+    /// Get possible heads of the issue
+    ///
+    /// Returns the head references from both the local repository and remotes
+    /// for this issue.
+    ///
+    pub fn heads(&self) -> Result<References<'r>> {
+        let glob = format!("**/dit/{}/head", self);
```

Comment:
It is meant to do just that. The rationale is that the implementation of `Display` always results in a unique string which may be used as part of reference names. This way, if we introduce "scoping" or "namespacing" for issues, the scope or namespace finds its way into references automatically.

We also could have a function which explicitly returns a string referring to the issue, which is specified to be used in ref-names and have the `Display` implementation rely on that function instead. If you insist, that is.